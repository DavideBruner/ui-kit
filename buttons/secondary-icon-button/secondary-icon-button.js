import React from 'react';
import PropTypes from 'prop-types';
import withMouseOverState from '../../hocs/with-mouse-over-state';
import AccessibleButton from '../accessible-button';
import styles from './secondary-icon-button.mod.css';

export const SecondaryIconButton = props => {
  let iconTheme = 'black';
  if (props.isDisabled) iconTheme = 'grey';
  else if (props.isMouseOver) iconTheme = 'green';
  return (
    <div
      onMouseOver={props.handleMouseOver}
      onMouseOut={props.handleMouseOut}
      className={styles.button}
    >
      <AccessibleButton
        label={props.label}
        onClick={props.onClick}
        isDisabled={props.isDisabled}
      >
        <div className={styles['icon-container']}>
          {React.cloneElement(props.icon, { theme: iconTheme })}
        </div>
      </AccessibleButton>
    </div>
  );
};

SecondaryIconButton.displayName = 'SecondaryIconButton';
SecondaryIconButton.propTypes = {
  label: PropTypes.string.isRequired,
  icon: PropTypes.element.isRequired,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,

  // HoC
  isMouseOver: PropTypes.bool.isRequired,
  handleMouseOver: PropTypes.func.isRequired,
  handleMouseOut: PropTypes.func.isRequired,
};
SecondaryIconButton.defaultProps = {
  isDisabled: false,
};

export default withMouseOverState(SecondaryIconButton);
