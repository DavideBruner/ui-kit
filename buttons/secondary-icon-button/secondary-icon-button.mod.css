.button {
  display: inline-flex;
  align-items: center;
  border: none;
  background: none;
  padding: 0;
  min-height: initial;
  cursor: pointer;
}

.icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
}
