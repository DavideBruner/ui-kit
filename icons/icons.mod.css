@import '../materials/colors.mod.css';

.theme-black {
  fill: var(--new-color-black);
}

.theme-grey {
  fill: var(--new-color-gray-60);
}

.theme-white {
  fill: var(--new-color-white);
}

.theme-blue {
  fill: var(--new-color-blue);
}

.theme-green {
  fill: var(--new-color-green-25);
}

.theme-apple-green {
  fill: var(--new-color-green);
}

.theme-orange {
  fill: var(--new-color-orange);
}

.theme-red {
  fill: var(--new-color-red);
}

svg.scale:not(:root) {
  width: 100%;
  height: auto;
}
