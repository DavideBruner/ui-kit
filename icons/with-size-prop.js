import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: size === 'small' ? 16 : 24,
  height: size === 'small' ? 16 : 24,
}));
