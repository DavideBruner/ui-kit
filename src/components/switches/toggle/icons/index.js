import ToggleButtonIcon from './toggle-button-default.svg';
import ToggleBarIcon from './toggle-bar-default.svg';

export default {
  ButtonDefault: ToggleButtonIcon,
  BarDefault: ToggleBarIcon,
};
