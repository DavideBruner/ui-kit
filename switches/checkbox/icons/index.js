// NOTE: explicit names for component names
import CheckboxDefaultIcon from './checkbox-default.svg';
import CheckboxDefaultHoverIcon from './checkbox-hover.svg';
import CheckboxDefaultErrorIcon from './checkbox-error.svg';
import CheckboxCheckedIcon from './checkbox-checked.svg';
import CheckboxCheckedHoverIcon from './checkbox-checked-hover.svg';
import CheckboxCheckedErrorIcon from './checkbox-checked-error.svg';
import CheckboxDisabledIcon from './checkbox-disabled.svg';
import CheckboxCheckedDisabledIcon from './checkbox-checked-disabled.svg';
import CheckboxIndeterminateIcon from './checkbox-indeterminate.svg';
import CheckboxIndeterminateHoverIcon from './checkbox-indeterminate-hover.svg';
import CheckboxIndeterminateDisabledIcon from './checkbox-indeterminate-disabled.svg';
import CheckboxIndeterminateErrorIcon from './checkbox-indeterminate-error.svg';

export default {
  Default: CheckboxDefaultIcon,
  Hover: CheckboxDefaultHoverIcon,
  Error: CheckboxDefaultErrorIcon,
  Checked: CheckboxCheckedIcon,
  CheckedHover: CheckboxCheckedHoverIcon,
  CheckedError: CheckboxCheckedErrorIcon,
  Disabled: CheckboxDisabledIcon,
  CheckedDisabled: CheckboxCheckedDisabledIcon,
  Indeterminate: CheckboxIndeterminateIcon,
  IndeterminateHover: CheckboxIndeterminateHoverIcon,
  IndeterminateDisabled: CheckboxIndeterminateDisabledIcon,
  IndeterminateError: CheckboxIndeterminateErrorIcon,
};
