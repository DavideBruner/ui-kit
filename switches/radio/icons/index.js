// NOTE: explicit names for component names
import RadioOptionDefaultIcon from './radio-option-default.svg';
import RadioOptionDefaultHoverIcon from './radio-option-hover.svg';
import RadioOptionCheckedIcon from './radio-option-checked.svg';
import RadioOptionCheckedHoverIcon from './radio-option-checked-hover.svg';
import RadioOptionDisabledIcon from './radio-option-disabled.svg';
import RadioOptionCheckedDisabledIcon from './radio-option-checked-disabled.svg';

export default {
  Default: RadioOptionDefaultIcon,
  Hover: RadioOptionDefaultHoverIcon,
  Checked: RadioOptionCheckedIcon,
  CheckedHover: RadioOptionCheckedHoverIcon,
  Disabled: RadioOptionDisabledIcon,
  CheckedDisabled: RadioOptionCheckedDisabledIcon,
};
