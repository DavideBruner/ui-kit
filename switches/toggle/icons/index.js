import withSizeProp from '../with-size-prop';
import ToggleBackgroundDefaultIcon from './toggle-normal-bkgr.svg';
import ToggleBackgroundHoverIcon from './toggle-active-hover-bkgr.svg';
import ToggleBackgroundDisabledIcon from './toggle-disable-bkgr.svg';
import ToggleBackgroundCheckedIcon from './toggled-bkgr.svg';
import ToggleBackgroundCheckedHoverIcon from './toggled-active-hover-bkgr.svg';
import ToggleBackgroundCheckedDisabledIcon from './toggled-disable-bkgr.svg';
import ToggleButtonDefaultIcon from './toggle-normal-bt.svg';
import ToggleButtonHoverIcon from './toggle-active-hover-bt.svg';
import ToggleButtonDisabledIcon from './toggle-disable-bt.svg';
import ToggleButtonCheckedIcon from './toggled-bt.svg';
import ToggleButtonCheckedHoverIcon from './toggled-active-hover-bt.svg';
import ToggleButtonCheckedDisabledIcon from './toggled-disable-bt.svg';

export default {
  BackgroundDefault: ToggleBackgroundDefaultIcon,
  BackgroundHover: ToggleBackgroundHoverIcon,
  BackgroundDisabled: ToggleBackgroundDisabledIcon,
  BackgroundChecked: ToggleBackgroundCheckedIcon,
  BackgroundCheckedHover: ToggleBackgroundCheckedHoverIcon,
  BackgroundCheckedDisabled: ToggleBackgroundCheckedDisabledIcon,
  ButtonDefault: withSizeProp(ToggleButtonDefaultIcon),
  ButtonHover: withSizeProp(ToggleButtonHoverIcon),
  ButtonDisabled: withSizeProp(ToggleButtonDisabledIcon),
  ButtonChecked: withSizeProp(ToggleButtonCheckedIcon),
  ButtonCheckedHover: withSizeProp(ToggleButtonCheckedHoverIcon),
  ButtonCheckedDisabled: withSizeProp(ToggleButtonCheckedDisabledIcon),
};
