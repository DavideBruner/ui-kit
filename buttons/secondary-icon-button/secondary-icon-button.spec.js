import React from 'react';
import { shallow } from 'enzyme';
import { FilterIcon } from '../../icons';
import { SecondaryIconButton } from './secondary-icon-button';

const createTestProps = custom => ({
  label: 'Accessible button',
  onClick: jest.fn(),
  icon: <FilterIcon />,

  // HoC
  isMouseOver: false,
  handleMouseOver: jest.fn(),
  handleMouseOut: jest.fn(),

  ...custom,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<SecondaryIconButton {...props} />);
  });

  it('outputs correct tree', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccessibleButton>', () => {
    expect(wrapper).toRender('AccessibleButton');
  });

  it('should render the icon', () => {
    expect(wrapper).toRender('FilterIcon');
  });

  it('renders an icon with black theme', () => {
    expect(wrapper.find('FilterIcon')).toHaveProp('theme', 'black');
  });

  describe('if mouse is over', () => {
    beforeEach(() => {
      props = createTestProps({ isMouseOver: true });
      wrapper = shallow(<SecondaryIconButton {...props} />);
    });
    it('renders an icon with green theme', () => {
      expect(wrapper.find('FilterIcon')).toHaveProp('theme', 'green');
    });
  });

  describe('if disabled', () => {
    beforeEach(() => {
      props = createTestProps({ isDisabled: true });
      wrapper = shallow(<SecondaryIconButton {...props} />);
    });
    it('renders a <AccessibleButton> with disabled prop', () => {
      expect(wrapper.find('AccessibleButton')).toHaveProp('isDisabled', true);
    });
    it('renders an icon with grey theme', () => {
      expect(wrapper.find('FilterIcon')).toHaveProp('theme', 'grey');
    });
  });
});

describe('callbacks', () => {
  let wrapper;
  let props;
  describe('onClick', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<SecondaryIconButton {...props} />);
      wrapper.find('AccessibleButton').prop('onClick')();
    });

    it('should invoke onClick', () => {
      expect(props.onClick).toHaveBeenCalledTimes(1);
    });
  });
});
