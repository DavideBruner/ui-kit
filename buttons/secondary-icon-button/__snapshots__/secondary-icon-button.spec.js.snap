// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering outputs correct tree 1`] = `
<div
  className="button"
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <AccessibleButton
    buttonAttributes={Object {}}
    isDisabled={false}
    isToggleButton={false}
    label="Accessible button"
    onClick={[MockFunction]}
  >
    <div
      className="icon-container"
    >
      <FilterIcon
        theme="black"
      />
    </div>
  </AccessibleButton>
</div>
`;
